packages/api/src/controllers/signing-key.ts

```diff
@@ -0,0 +1,238 @@
+    const [output, newCursor] = await db.signingKey.find(query, {
+      limit,
+      cursor,
+      fields,
+      from,
+      order: parseOrder(fieldsMap, order),
+      process: ({ data, usersdata, count: c }) => {
+        if (count) {
+          res.set("X-Total-Count", c);
+        }
+        return {
+          data,
+          user: db.user.cleanWriteOnlyResponse(usersdata),
+        };
+      },
+    });
+
+    res.status(200);
+
+    if (output.length > 0 && newCursor) {
+      res.links({ next: makeNextHREF(req, newCursor) });
+    }
+    return res.json(output);
+  }
+
+  const query = parseFilters(fieldsMap, filters);
+  query.push(sql`signing_key.data->>'userId' = ${req.user.id}`);
+  query.push(sql`signing_key.data->>'deleted' IS NULL`);
+
+  let fields = " signing_key.id as id, signing_key.data as data";
+  if (count) {
+    fields = fields + ", count(*) OVER() AS count";
+  }
+  const from = `signing_key`;
+  const [output, newCursor] = await db.signingKey.find(query, {
+    limit,
+    cursor,
+    fields,
+    from,
+    order: parseOrder(fieldsMap, order),
+    process: ({ data, count: c }) => {
+      if (count) {
+        res.set("X-Total-Count", c);
+      }
+      return data;
+    },
+  });
+
+  res.status(200);
+
+  if (output.length > 0) {
+    res.links({ next: makeNextHREF(req, newCursor) });
+  }
+
+  return res.json(output);
+});
+
+app.get("/:id", authorizer({}), async (req, res) => {
+  const signingKey = await db.signingKey.get(req.params.id);
+  if (!signingKey || signingKey.deleted) {
+    res.status(404);
+    return res.json({
+      errors: ["not found"],
+    });
+  }
+
+  if (req.user.admin !== true && req.user.id !== signingKey.userId) {
+    res.status(403);
+    return res.json({
+      errors: ["user can only request information on their own signingKeys"],
+    });
+  }
+
+  res.json(signingKey);
+});
+
+app.post(
+  "/",
+  validatePost("new-signing-key-payload"),
+  authorizer({}),
+  async (req, res) => {
+    const query = parseFilters(fieldsMap, "");
+    query.push(sql`signing_key.data->>'userId' = ${req.user.id}`);
+    query.push(sql`signing_key.data->>'deleted' IS NULL`);
+    const [output] = await db.signingKey.find(query, {
+      limit: 100,
+      fields: " signing_key.id as id, signing_key.data as data",
+      from: `signing_key`,
+      order: parseOrder(fieldsMap, "createdAt-true"),
+      process: ({ data }) => {
+        return data;
+      },
+    });
+
+    if (output.length > 10) {
+      res.status(403);
+      return res.json({
+        errors: ["user can only have up to 10 signing keys, delete some first"],
+      });
+    }
+
+    const id = uuid();
+    const keypair = generateSigningKeys();
+
+    if (!req.body.name || req.body.name === "") {
+      req.body.name = "Signing Key " + (output.length + 1);
+    }
+
+    var doc: WithID<SigningKey> = {
+      id,
+      name: req.body.name,
```

Comment:
We should avoid mutating objects, especially the ones from the HTTP framework. This works as well and (IMO) is cleaner/clearer. WDYT?
```suggestion
    var doc: WithID<SigningKey> = {
      id,
      name: req.body.name || ("Signing Key " + (output.length + 1)),
```

Otherwise we could just have a helper variable in this function as well for the mutable data. Like
```ts
let name = req.body.name
if (!name) {
  name = "..."
}
// use name
```
(`!str` already checks both that the stream is defined and not empty! since the empty string is fals-y, it also works on the boolean operator `||` to fallback to a default)

Reply:
Makes much more sense than mutating the req body, I added this